src/Docker/Client/Internal.hs

```diff
@@ -106,3 +106,6 @@ getEndpointContentType :: Endpoint -> BSC.ByteString
 getEndpointContentType (BuildImageEndpoint _ _) = BSC.pack "application/tar"
 getEndpointContentType _ = BSC.pack "application/json; charset=utf-8"
 
+getEndpointTimeout :: Endpoint -> HTTP.ResponseTimeout 
+getEndpointTimeout (WaitContainerEndpoint _) = HTTP.responseTimeoutNone
+getEndpointTimeout _ = HTTP.responseTimeoutDefault
```

Comment:
Wait, doesn't this render the `NOTE` above irrelevant?

Reply:
Thanks for reviewing this! The third commit does make the `NOTE` from the first irrelevant, but it is a question of whether you want `getEndpointTimeout` in the code or not. Having it always forces no timeout for the endpoint and no way for the user of the library to override this. Specifying different `ManagerSettings` and not merging the third commit allows users to specify a timeout themselves, but it is then shared between all endpoints. 

Personally, I have been using a version of this library with my changes in production for the past year and the approach with `getEndpointTimeout` is fine since you can always wrap the call to `waitContainer` in a timeout yourself. Happy to remove the `NOTE` if you are happy to merge everything otherwise.